integrations/nvidia/src/haystack_integrations/components/rankers/nvidia/ranker.py

```diff
@@ -0,0 +1,204 @@
+import warnings
+from typing import Any, Dict, List, Literal, Optional
+
+from haystack import Document, component, default_from_dict, default_to_dict
+from haystack.utils import Secret, deserialize_secrets_inplace
+
+from haystack_integrations.utils.nvidia import NimBackend, url_validation
+
+_DEFAULT_MODEL = "nvidia/nv-rerankqa-mistral-4b-v3"
+
+_MODEL_ENDPOINT_MAP = {
+    "nvidia/nv-rerankqa-mistral-4b-v3": "https://ai.api.nvidia.com/v1/retrieval/nvidia/nv-rerankqa-mistral-4b-v3/reranking",
+}
+
+
+@component
+class NvidiaRanker:
+    """
+    A component for ranking documents using ranking models provided by
+    [NVIDIA NIMs](https://ai.nvidia.com).
+
+    Usage example:
+    ```python
+    from haystack_integrations.components.rankers.nvidia import NvidiaRanker
+    from haystack import Document
+    from haystack.utils import Secret
+
+    ranker = NvidiaRanker(
+        model="nvidia/nv-rerankqa-mistral-4b-v3",
+        api_key=Secret.from_env_var("NVIDIA_API_KEY"),
+    )
+    ranker.warm_up()
+
+    query = "What is the capital of Germany?"
+    documents = [
+        Document(content="Berlin is the capital of Germany."),
+        Document(content="The capital of Germany is Berlin."),
+        Document(content="Germany's capital is Berlin."),
+    ]
+
+    result = ranker.run(query, documents, top_k=2)
+    print(result["documents"])
+    ```
+    """
+
+    def __init__(
+        self,
+        model: Optional[str] = None,
+        truncate: Optional[Literal["NONE", "END"]] = None,
+        api_url: Optional[str] = None,
+        api_key: Optional[Secret] = None,
+        top_k: int = 5,
+    ):
+        """
+        Create a NvidiaRanker component.
+
+        :param model:
+            Ranking model to use.
+        :param truncate:
+            Truncation strategy to use. Can be "NONE" or "END". Defaults to NIM's default.
+        :param api_key:
+            API key for the NVIDIA NIM.
+        :param api_url:
+            Custom API URL for the NVIDIA NIM.
+        :param top_k:
+            Number of documents to return.
+        """
+        if model is not None and not isinstance(model, str):
+            msg = "Ranker expects the `model` parameter to be a string."
+            raise TypeError(msg)
+        if not isinstance(api_url, (str, type(None))):
+            msg = "Ranker expects the `api_url` parameter to be a string."
+            raise TypeError(msg)
+        if truncate is not None and truncate not in ["NONE", "END"]:
+            msg = "Invalid truncate value. Must be 'NONE' or 'END'."
+            raise ValueError(msg)
+        if not isinstance(top_k, int):
+            msg = "Ranker expects the `top_k` parameter to be an integer."
+            raise TypeError(msg)
+
+        # todo: detect default in non-hosted case (when api_url is provided)
```

Comment:
Is this not something we can handle in this PR?

Reply:
i would prefer to do this in a follow up feat, if that's ok with you

Reply:
Works for me.